src/main/java/com/yelp/nrtsearch/server/luceneserver/innerhit/InnerHitContext.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.innerhit;
+
+import com.yelp.nrtsearch.server.luceneserver.QueryNodeMapper;
+import com.yelp.nrtsearch.server.luceneserver.doc.DefaultSharedDocContext;
+import com.yelp.nrtsearch.server.luceneserver.field.FieldDef;
+import com.yelp.nrtsearch.server.luceneserver.highlights.HighlightFetchTask;
+import com.yelp.nrtsearch.server.luceneserver.search.FetchTasks;
+import com.yelp.nrtsearch.server.luceneserver.search.SearchContext;
+import com.yelp.nrtsearch.server.luceneserver.search.collectors.DocCollector;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.join.BitSetProducer;
+import org.apache.lucene.search.join.QueryBitSetProducer;
+
+public class InnerHitContext extends SearchContext {
```

Comment:
Does this really need to extend `SearchContext`? If it needs to retrieve document fields, it can just implement the `FieldFetchContext` interface.

Reply:
My only doubt is that do we want to put the rootSearchContext for fetchTask.processHit, or the "parentContext" in case of multi-layer innerHit.

It looks like be out of the current concern now. Change to FieldFetchContext makes sense to me.

Reply:
Forget there were dead loop issue without this design (extends searchContext).

When doFillTask, it will get searchContext and execute all fetchtasks (include innerHit). If we set the searchContext to the rootSearchContext in the innerHit, innerHit's doFillTask will find all rootFetchTasks, and execute all fetchTasks again, which includes itself. So infitnie loop is formed.

I will keep the design, and let getSearchContext to return itself for innerHit. This sounds like a more correct way.

Reply:
A way to avoid that situation is move `getHighlightFetchTask()` and `getInnerHitFetchTasks()` into the `FieldFetchContext` interface. That is already the case for the standard `FetchTasks`.

Reply:
Thanks for the suggestion. I add the highlight and innerHit into the FetchTasks